pkg/kv/kvpb/api.proto

```diff
@@ -71,6 +71,29 @@ enum RoutingPolicy {
   NEAREST = 1;
 }
 
+// ConnectionClass specifies the preferred RPC connection class to use for
+// sending or retransmitting the BatchRequest.
+//
+// This enum is similar and related to rpc.ConnectionClass, but the semantics
+// differ. This enum only defines classes used for BatchRequest traffic, and
+// only defines preferences. There are more classes in rpc package that are not
+// relevant here, such as RaftClass.
+enum ConnectionClass {
+  // DEFAULT declares no preference.
+  //
+  // Note that this does not necessarily correspond to rpc.DefaultClass. The
+  // actual connection class is chosen based on various factors. For example,
+  // for some critical system ranges, requests use rpc.SystemClass instead.
+  DEFAULT = 0;
+  // RANGEFEED declares a preference for RangefeedClass.
+  //
+  // As an example, at the time of writing this is used by changefeed initial
+  // scan requests. The rangefeed traffic is potentially bulky, so it has its
+  // dedicated connection class. Changefeed is a major consumer of rangefeed,
+  // and has similar properties, so we place them together.
+  RANGEFEED = 2;
```

Comment:
Should we perhaps call this something else? Like `BULK_DATA` or similar? It's not specific to rangefeed -- we now use that for scan requests for example.

Reply:
I like the idea. It would be a bit ugly though because this translates to `rpc.RangefeedClass`. Maybe we should generalize/deprecate `rpc.RangefeedClass` first.

@erikgrinaker WDYT?

Reply:
Added a commit that renames the `rpc` class too.

Reply:
This seems misleading to me, since we don't actually use it for bulk traffic -- imports, schema changes, restores etc. don't use it. Adding a separate bulk class might make sense if we're actually going to use it, but I don't think we should spend time on it now, and I'm not sure it makes sense to send rangefeeds across the bulk class even if we have one (some customers may not want bulk traffic to increase feed latency, nor do we for our internal system rangefeeds).

I'd be inclined to just keep `RangefeedClass` for these, since that's the only thing it will be used for in 24.1 (fine, initial scans aren't rangefeeds, but initial scans are only used together with rangefeeds so that seems fine to me).

Reply:
Reverted to using the rangefeed class. We'll generalize later if needed.